lib/ansible/module_utils/facts/network/linux.py

```diff
@@ -147,9 +147,13 @@ def get_interfaces_info(self, ip_path, default_ipv4, default_ipv6):
                 interfaces[device]['perm_macaddress'] = get_file_content(os.path.join(path, 'bonding_slave', 'perm_hwaddr'), default='')
             if os.path.exists(os.path.join(path, 'device')):
                 interfaces[device]['pciid'] = os.path.basename(os.readlink(os.path.join(path, 'device')))
+            if os.path.exists(os.path.join(path, 'duplex')):
+                duplex = get_file_content(os.path.join(path, 'duplex'))
+                if duplex is not None and duplex != 'unknown':
+                    interfaces[device]['duplex'] = duplex
             if os.path.exists(os.path.join(path, 'speed')):
                 speed = get_file_content(os.path.join(path, 'speed'))
-                if speed is not None:
+                if speed is not None and duplex is not None and duplex != 'unknown':
```

Comment:
hi @alikins 
I cant agree more with u about the setup module just need output the most common basic data. But i think the accuracy of the data is also important.
In my case, when the duplex display "unknown", the speed will display wrong, it will display like "65535" or a larger number. At this time, i think not show better than show a wrong data even if this error is not caused by our query.